hnn_core/optimization.py

```diff
@@ -351,24 +439,29 @@ def optimize_evoked(params, target_dpl, initial_dpl, maxiter=50,
     smooth_window_len : int
         The length of the hamming window (in samples) to smooth the
         simulated dipole waveform in each optimization step.
+    dt : float
+        The integration time step (ms) of h.CVode during simulation.
 
     Returns
     -------
-    params : dict
-        The optimized params dictionary.
+    net : Instance of Network object
```

Comment:
I think the only issue here is we don't have a good way to save net objects. Since running optimization is pretty expensive, what would be the alternative? Pickle the drive dictionary?

Reply:
This is perfect timing actually, I was just about to ping you and @jasmainak.

I totally agree that saving the `Network` instance is an issue. Here are a few thoughts to consider:

- `optimize_evoked()` now operates similarly to the simulation engine, `simulate_dipole()`, where the user first defines a network and then passes it into this function to undergo simulation / simulation-based optimization. This makes optimization compatible with any of our network templates. In my opinion, this is consistent with our current API where the user can define any `Network` instance and trust that it can be simulated. Furthermore, it ensures that the `Network` object, upon which drive optimization is dependent, is exactly the same object defined explicitly by the user. Should they decide to create a custom `Network` object that complies with our API, it can be optimized as well.
- If we're wanting to eventually do away with .param files, we need to remove dependance on them. This has been accomplished here by using the `_get_drive_params()` function (which is private for now, though we can consider eventually making it public) and then setting the drives during optimization using explicit calls to the API. This makes it easier to follow which evoked drive parameters are being optimized over (e.g., static parameters like the number of drive cells aren't explored, while dynamic parameters like the mean drive time are), is more flexible, and ensures that any bugs that emerge are a result of our API, rather than some obscure issue saving, loading, or parsing .param files.
- To address the issue you raise here, we can make a follow-up PR where we refine `_get_drive_params()` and maybe move it to `Network` so that users can extract and view drive parameters more easily (and plot them should they desire).